src/core/gui/PagePreviewDecoration.cpp

```diff
@@ -0,0 +1,96 @@
+#include "PagePreviewDecoration.h"
+
+#include <math.h>
+
+#include "control/Control.h"
+#include "control/settings/Settings.h"
+#include "util/raii/CairoWrappers.h"
+
+void PagePreviewDecoration::drawDecoration(cairo_t* cr, SidebarPreviewPageEntry* pageEntry, Control* control) {
+    switch ( control->getSettings()->getSidebarNumberingStyle()) {
+    case SidebarNumberingStyle::SIDEBAR_NUM_STYLE_1:
+        PagePreviewDecoration::drawStyle1(cr, pageEntry, control);
+        break;
+    case SidebarNumberingStyle::SIDEBAR_NUM_STYLE_2:
+        PagePreviewDecoration::drawStyle2(cr, pageEntry, control);
+        break;
+    case SidebarNumberingStyle::SIDEBAR_NUM_STYLE_3:
+        PagePreviewDecoration::drawStyle3(cr, pageEntry, control);
+        break;
+    default:
+        break;
+    }
+
+}
+
+void PagePreviewDecoration::drawStyle1(cairo_t* cr, SidebarPreviewPageEntry* pageEntry, Control* control) {
+    xoj::util::CairoSaveGuard saveGuard(cr);
+    cairo_text_extents_t extents;
+    std::string pageNumber = std::to_string(pageEntry->getIndex() + 1);
+    Color color = control->getSettings()->getBackgroundColor();
+    if (pageEntry->isSelected()) {
+        color = control->getSettings()->getBorderColor();
+    }
+    Util::cairo_set_source_rgbi(cr, color);
+    cairo_select_font_face(cr, "Sans", CAIRO_FONT_SLANT_NORMAL, CAIRO_FONT_WEIGHT_NORMAL);
+    cairo_set_font_size(cr, FONT_SIZE);
+    cairo_text_extents(cr, pageNumber.c_str(), &extents);
+    double x = pageEntry->getWidth() / 2 - (extents.width / 2 + extents.x_bearing);
+    double y = pageEntry->getHeight() - 5 - (extents.height / 2 + extents.y_bearing);
```

Comment:
nit: the `5` should be moved into a named constant to help document what it's for, e.g. `const double MARGIN_BOTTOM = 5.0`